packages/chain-adapters/src/utils/utxoUtils.ts

```diff
@@ -0,0 +1,64 @@
+import { BTCInputScriptType } from '@shapeshiftoss/hdwallet-core'
+import { Asset, ChainTypes } from '@shapeshiftoss/types'
+import { BIP32Params } from '@shapeshiftoss/types'
+import { ChainAdapter as BitcoinChainAdapter } from '../bitcoin/BitcoinChainAdapter'
+import { ChainAdapter as EthereumChainAdapter } from '../ethereum/EthereumChainAdapter'
+/**
+ * Utility function to bip32Params from script type
+ * Get return an object with bip32Params derived from script type and asset
+ * @param scriptType
+ * @param asset
+ * @returns BIP32Params
+ */
+export const bip32FromScript = (
+  scriptType: BTCInputScriptType,
+  asset: Asset
+): BIP32Params => {
+  const purpose = purposeFromScript(scriptType)
+
+  switch(asset.chain) {
+    case ChainTypes.Bitcoin: {
+      return { ...BitcoinChainAdapter.defaultBIP32Params, purpose }
+    }
+    case ChainTypes.Ethereum: {
+      return { ...EthereumChainAdapter.defaultBIP32Params, purpose }
+    }
+  }
+}
+
+/**
+ * Returns coin purpose from a given script type
+ * @param scriptType
+ * @returns 44 | 49 | 84
+ */
+export const purposeFromScript = (
+  scriptType: BTCInputScriptType
+): 44 | 49 | 84 => {
+  switch (scriptType) {
+    case BTCInputScriptType.SpendP2SHWitness:
+      return 49
+    case BTCInputScriptType.SpendAddress:
+      return 44
+    case BTCInputScriptType.SpendWitness:
+    default:
+      return 84
+  }
+}
+
+/**
+ * Utility function to get params to get BIP32Params and scriptType for chain-adapter functions (getAddress, buildSendTransaction)
+ * Get return an object with utxo bip32Params derived from script type and asset
+ * @param scriptType
+ * @param asset
+ * @returns object with BIP32Params and scriptType or undefined
+ */
+export const bip32AndScript = (
```

Comment:
I'm not sure what problem you're trying to solve here, but I don't think it should be solved this way.
- there's no need to return the script type if the caller knows it already
- you shouldn't get bip32params from an asset; you might be able to get a _default_ bip32 path from an asset, but an asset may be stored at any other path, and in the case of change addresses _definitely is_ in the normal use case.

Reply:
I return the script thype so I can easily spread it and bip32Params into chain-adapter calls  in on easy line where its needed. namely getAddress and buildSendTx.